app.py

```diff
@@ -218,11 +218,15 @@
                     justification = adviser_result["report"]["products"][0]["justification"]
 
                     report = json.dumps(adviser_report, indent=2)
-                    if len(report) > MAX_CHARACTERS_LENGTH:
-                        reduced_report = adviser_report["pipeline"]
-                        report = json.dumps(reduced_report, indent=2)
                     # a hack to display indentation spaces in the resulting HTML
                     report = re.sub("\n {2,}", lambda m: "\n" + "&ensp;" * (len(m.group().strip("\n"))), report,)
+                    _LOGGER.info("on_thamos_workflow_finished: len(report)=%s", len(report))
+                    # TODO: Split report results to include only relevant information
+                    if len(report) > MAX_CHARACTERS_LENGTH:
+                        reduced_report = adviser_report["metadata"]
```

Comment:
Not sure if I read correctly, but I think you want something like:

```python
reduced_report.pop("metadata", None)
```

Reply:
`adviser_report` does not have `metadata` key.
```
adviser_payload: dict = await response.json() where response is the endpoint on user-api

adviser_result: dict = adviser_payload["result"]

adviser_report: dict = adviser_result["report"]

reduced_report=?
```

Moreover justification has two keys now I see:

```
Thoth's adviser finished with conclusion: 'success'

Justification:
[{'message': "No observations spotted for this stack in Thoth's database", 'type': 'INFO'}]

See the report below for more details.
```

so I will add 
`justification = adviser_result["report"]["products"][0]["justification"]["message"]`

Reply:
> `adviser_report` does not have `metadata` key.
> 
> ```
> adviser_payload: dict = await response.json() where response is the endpoint on user-api
> 
> adviser_result: dict = adviser_payload["result"]
> 
> adviser_report: dict = adviser_result["report"]
> 
> reduced_report=?
> ```
> 
> Moreover justification has two keys now I see:
> 
> ```
> Thoth's adviser finished with conclusion: 'success'
> 
> Justification:
> [{'message': "No observations spotted for this stack in Thoth's database", 'type': 'INFO'}]
> 
> See the report below for more details.
> ```
> 
> so I will add
> `justification = adviser_report["products"][0]["justification"]["message"]`

Does it mean the stripping will be done on `adviser_payload["result"]`?

Reply:
The interesting information to provide to the users are in the report no? The entire document can be still be checked using the link.

Reply:
Yes, the relevant entries:

```
result.report.products[0].advised_runtime_environment
result.report.products[0].justification
result.report.products[0].project.requirements
result.report.products[0].project.requirements_locked
result.report.stack_info
```

For error reports on adviser side:
```
result.error_msg
```

Others can be discarded.

Reply:
then in the `report` I would give `result.report.products[0]` and pop `justification` not to be redundant.

Reply:
what about `reduced one`?

Reply:
Do you mean the removed ones? I think they don't need to show up on github - they can be retrieved from the UI service (posting a link in report present on GitHub, if possible, might be good enough).

Reply:
```
result.report.products[0].advised_runtime_environment
result.report.stack_info
```

Reply:
> Do you mean the removed ones? I think they don't need to show up on github - they can be retrieved from the UI service (posting a link in report present on GitHub, if possible, might be good enough).

No I mean if we hit `GitHub limit` I have to cut out some things.

Reply:
Oh, if we hit github limit just with these few, then that might be a problem to report back recommendations.